### PR TITLE
[Middleware_02] Define MiddlewareCommInterfaceImplementationTest

### PR DESCRIPTION
**Pull request by NMoua-Edu**
Working on #59  - Define MiddlewareCommInterfaceImplementationTest class

### CODE REVIEW

File: docs/uml-class-diagrams/middleware02/test/MiddlewareCommInterfaceImplementationTests/MiddlewareCommInterfaceImplementationTests.puml [hunk 1]
```diff
@@ -2,7 +2,8 @@
 
 package edu.oakland.test.middleware02 {
     class MiddlewareCommInterfaceImplementationTests {
-
+        + void RfidInIsRfidOut()
+
```
Comment (andrewdimmer) [inline docs/uml-class-diagrams/middleware02/test/MiddlewareCommInterfaceImplementationTests/MiddlewareCommInterfaceImplementationTests.puml]: Fix the casing --> Methods in Java are always in camelCase.
Also, make sure to have a test case to verify that the `MiddlewareCommLinkManger` is not null!
```suggestion
        + void rfidInIsRfidOut()
        + void middlewareCommLinkManagerIsNotNull()
```

File: src/main/java/edu/oakland/production/middleware02/MiddlewareCommInterface.java [hunk 2]
```diff
@@ -5,5 +5,4 @@
 public interface MiddlewareCommInterface {
```
Comment (andrewdimmer) [inline src/main/java/edu/oakland/production/middleware02/MiddlewareCommInterface.java:5]: For your sanity in terms of getting this pull request merged, and my sanity in terms of keeping track of design vs. implementation, can we move this an the other Java files to another PR? You should be able to cut them to another folder (outside the repo), commit the change, the set up a new branch and paste them in there to remove them from this PR.

File: src/main/java/edu/oakland/production/middleware02/MiddlewareCommInterfaceImplementation.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package edu.oakland.production.middleware02;
```
Comment (andrewdimmer) [inline src/main/java/edu/oakland/production/middleware02/MiddlewareCommInterfaceImplementation.java]: For your sanity in terms of getting this pull request merged, and my sanity in terms of keeping track of design vs. implementation, can we move this an the other Java files to another PR? You should be able to cut them to another folder (outside the repo), commit the change, the set up a new branch and paste them in there to remove them from this PR.

File: src/test/java/edu/oakland/test/middleware02/RequestRfidTest.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package edu.oakland.test.middleware02;
```
Comment (andrewdimmer) [inline src/test/java/edu/oakland/test/middleware02/RequestRfidTest.java]: For your sanity in terms of getting this pull request merged, and my sanity in terms of keeping track of design vs. implementation, can we move this an the other Java files to another PR? You should be able to cut them to another folder (outside the repo), commit the change, the set up a new branch and paste them in there to remove them from this PR.

**Review by andrewdimmer — CHANGES_REQUESTED** [review]
One minor casing change, otherwise your design looks good.

For your sanity in terms of getting this pull request merged though, and my sanity in terms of keeping track of design vs. implementation, can we move all of the Java files to another PR? You should be able to cut them to another folder (outside the repo), commit the change, the set up a new branch and paste them in there to remove them from this PR.

**Review by andrewdimmer — APPROVED** [review]
Looks good to me, all set to merge!

**Review by kwillard34 — APPROVED** [review]

**Review by mckenzieking — APPROVED** [review]
